File: blobstore/sample code/python/sample_blob.py
from azure.storage.blob import BlobServiceClient


class BLOB:
    blob_service = None

    def __init__(self, endpoint, account_name, account_key):
        self.blob_service = BlobServiceClient.from_connection_string(
            'DefaultEndpointsProtocol=https;AccountName=' + account_name + ';AccountKey=' +
            account_key + ';EndpointSuffix=' + endpoint)

    # -------list containers---------------
    def list_container(self):
        containers = []
        results = self.blob_service.list_containers()
        for result in results:
            containers.append(result.name)
        return containers

    # -------create container---------------
    def create_container(self, container_name):
        self.blob_service.create_container(container_name)
        print("create container:", container_name)

    # -------delete container---------------
    def delete_container(self, container_name):
        self.blob_service.delete_container(container_name)
        print("delete container:", container_name)

    # -------list blobs---------------
    def list_blob(self, container_name):
        list = []
        container_client = self.blob_service.get_container_client(container_name)
        blobs = container_client.list_blobs()
        for b in blobs:
            list.append(b["name"])
        return list

    # -------upload blob---------------
    def upload_blob(self, container_name, blob_name, file):
        container_client = self.blob_service.get_container_client(container_name)
        container_client.upload_blob(blob_name, file)
        print("upload:", file)

    # -------download blob---------------
    def download_blob(self, container_name, blob_name):
        container_client = self.blob_service.get_container_client(container_name)
        file = container_client.download_blob(blob_name)
        return file

    # -------delete blob---------------
    def delete_blob(self, container_name, blob_name):
        container_client = self.blob_service.get_container_client(container_name)
        container_client.delete_blob(blob_name)
        print('delete blob:', blob_name)

File: blobstore/sample code/golang-new/golang/blob/sample_blob.go
package blob

import (
	"context"
	"fmt"
	"github.com/Azure/azure-storage-blob-go/azblob"
	"io"
	"log"
	"net/url"
	"simple_code/golang/properties"
	"strings"
)

type Blob struct {
	ServiceUrl *azblob.ServiceURL
}

//endpoint: blob.core.chinacloudapi.cn
//endpoint: blob.core.windows.net
func NewBlobClient(cre *properties.Credentials) (blob Blob, err error) {
	var rawUrl string
	if strings.HasPrefix(cre.Endpoint, "blob.") {
		rawUrl = fmt.Sprintf("https://%s.%s/", cre.AccessKey, cre.Endpoint)
	} else {
		rawUrl = fmt.Sprintf("https://%s.blob.%s/", cre.AccessKey, cre.Endpoint)
	}
	credential, err := azblob.NewSharedKeyCredential(cre.AccessKey, cre.Secretkey)
	if err != nil {
		//HandleError(err)
		return blob, err
	}
	uRL, _ := url.Parse(rawUrl)
	p := azblob.NewPipeline(credential, azblob.PipelineOptions{})
	serviceUrl := azblob.NewServiceURL(*uRL, p)
	blob.ServiceUrl = &serviceUrl
	return blob, err
}

func (blob *Blob) ListContainer() (containers []string, err error) {
	response, err := blob.ServiceUrl.ListContainersSegment(context.Background(), azblob.Marker{}, azblob.ListContainersSegmentOptions{})
	if err != nil {
		//HandleError(err)
		log.Println("err:", err)
		return nil, err
	}
	//fmt.Println(response, "---------------------------------------------------")
	if len(response.ContainerItems) > 0 {
		for _, item := range response.ContainerItems {
			containers = append(containers, item.Name)
		}
	}
	return containers, err
}

func (blob *Blob) CreateContainer(containerName string) error {
	containerUrl := blob.ServiceUrl.NewContainerURL(containerName)
	_, err := containerUrl.Create(context.Background(), azblob.Metadata{}, azblob.PublicAccessNone)
	if err != nil {
		//HandleError(err)
	}
	return err
}

func (blob *Blob) DeleteContainer(containerName string) error {
	containerUrl := blob.ServiceUrl.NewContainerURL(containerName)
	_, err := containerUrl.Delete(context.Background(), azblob.ContainerAccessConditions{})
	if err != nil {
		//HandleError(err)
	}
	return err
}

func (blob *Blob) ListBlob(containerName string) (blobs []string, err error) {
	containerUrl := blob.ServiceUrl.NewContainerURL(containerName)
	response, err := containerUrl.ListBlobsFlatSegment(context.Background(), azblob.Marker{}, azblob.ListBlobsSegmentOptions{})
	if err != nil {
		//HandleError(err)
	}
	if response != nil {
		for _, blob := range response.Segment.BlobItems {
			blobs = append(blobs, blob.Name)
		}
	}
	return blobs, err
}

//strings.NewReader("")
func (blob *Blob) UploadBlob(containerName, blobName string, body io.ReadSeeker) (error) {
	containerUrl := blob.ServiceUrl.NewContainerURL(containerName)
	blobUrl := containerUrl.NewBlockBlobURL(blobName)
	//ctx context.Context, body io.ReadSeeker, h BlobHTTPHeaders, metadata Metadata, ac BlobAccessConditions
	rsp, err := blobUrl.Upload(context.Background(), body, azblob.BlobHTTPHeaders{}, nil,
		azblob.BlobAccessConditions{}, azblob.DefaultAccessTier, azblob.BlobTagsMap{})
	if err != nil {
		//HandleError(err)
	}
	fmt.Println(rsp)
	return err
}

func (blob *Blob) DownloadBlob(containerName string, blobName string) (msg io.ReadCloser, err error) {
	containerUrl := blob.ServiceUrl.NewContainerURL(containerName)
	blobUrl := containerUrl.NewBlockBlobURL(blobName)
	rsp, err := blobUrl.Download(context.Background(), 0, 0, azblob.BlobAccessConditions{}, false)
	if err != nil {
		//HandleError(err)
		return nil, err
	}
	return rsp.Response().Body, err
}

func (blob *Blob) DeleteBlob(containerName string, blobName string) (error) {
	containerUrl := blob.ServiceUrl.NewContainerURL(containerName)
	blobUrl := containerUrl.NewBlockBlobURL(blobName)
	rsp, err := blobUrl.Delete(context.Background(), azblob.DeleteSnapshotsOptionNone, azblob.BlobAccessConditions{})
	if err != nil {
		//HandleError(err)
	}
	fmt.Println(rsp)
	return err
}
